### PR TITLE
ValueShuffle Hardening

### PR DESCRIPTION
**Pull request by dbmcclain**
cleanup, light refactoring, 
initial error traps and race conditions examined, 
first cut at uniform state handling on error returns

remove warning ignores and cleanup to standard Rust
remove unused code

### CODE REVIEW

File: wallet/src/lib.rs [hunk 1]
```diff
@@ -21,7 +21,7 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 
-#![deny(warnings)]
+// #![deny(warnings)]
```
Comment (rtsisyk) [inline wallet/src/lib.rs]: Please fix errors.

Comment (dbmcclain) [reply wallet/src/lib.rs]: what errors?

Comment (rtsisyk) [reply wallet/src/lib.rs]: Please revert `#![deny(warnings)]` and fix all warning instead.

Comment (dbmcclain) [reply wallet/src/lib.rs]: deny(warnings) reverted. Sorry, I was under the impression that this softened the Rust scrutiny and that leaving it out makes Rust more critical of small semantic errors. I guess I had it backwards? At any rate, reverted as per instructions and no errors, no warnings.

**Pull request merged**